boot/seal.go

```diff
@@ -95,40 +98,101 @@ func sealKeyToModeenv(key secboot.EncryptionKey, model *asserts.Model, modeenv *
 		bootloader.RoleRunMode:  bl.Name(),
 	}
 
-	// get model parameters from bootchains
-	modelParams, err := sealKeyModelParams(pbc, roleToBlName)
-	if err != nil {
-		return fmt.Errorf("cannot prepare for key sealing: %v", err)
-	}
 	// make sure relevant locations exist
 	for _, p := range []string{InitramfsEncryptionKeyDir, InstallHostFDEDataDir} {
 		if err := os.MkdirAll(p, 0755); err != nil {
 			return err
 		}
 	}
-	sealKeyParams := &secboot.SealKeyParams{
-		ModelParams:          modelParams,
-		KeyFile:              filepath.Join(InitramfsEncryptionKeyDir, "ubuntu-data.sealed-key"),
-		TPMPolicyAuthKeyFile: filepath.Join(InstallHostFDEDataDir, "tpm-policy-auth-key"),
-		TPMLockoutAuthFile:   filepath.Join(InstallHostFDEDataDir, "tpm-lockout-auth"),
+
+	// the boot chains we seal the fallback object to
+	rpbc := toPredictableBootChains(recoveryBootChains)
+
+	authKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
looks like secboot.SealKeyToTPM() generates and returns the, would it make sense to use it rather than generate one ourselves?

Reply:
Yes, I think we could get it from the first seal operation and reuse it in the second.

Reply:
In fact we can't because the auth key returned by the seal operation is a derivative of the supplied auth key (the elliptic curve private key). Maybe there's a way to get back to the original key from the returned data and if this is the case we can refactor later.